backend/benefit/applications/services/csv_export_base.py

```diff
@@ -0,0 +1,108 @@
+import csv
+import datetime
+import decimal
+import operator
+from dataclasses import dataclass
+from io import StringIO
+from typing import Any, Callable, Union
+
+from applications.enums import OrganizationType
+
+NO_DEFAULT_VALUE = object()
+
+
+@dataclass
+class CsvColumn:
+    heading: str
+    cell_data_source: Union[Callable, str]
+    formatter: Union[Callable, None] = None
+    default_value: Any = NO_DEFAULT_VALUE
+
+
+def get_organization_type(application):
+    return str(
+        OrganizationType.resolve_organization_type(application.company.company_form)
+    )
+
+
+def nested_queryset_attr(
+    related_name, queryset_idx, nested_attr_name, default_value=""
+):
+    """
+    Return a function that:
+    * takes an "item" object as a parameter
+      "item" should be an object with an RelatedManager attribute named related_name
+    * returns item.attr_name[idx].nested_attr_name
+    * In case the attribute is missing, return the default value
+    * dotted attribute access (like "a.b" is supported for attr_name and nested_attr_name
+    """
+
+    def getter(item):
+        try:
+            nested_obj = operator.attrgetter(related_name)(item).all()[queryset_idx]
+            return operator.attrgetter(nested_attr_name)(nested_obj)
+        except (AttributeError, IndexError):
+            return default_value
+
+    return getter
+
+
+class CsvExportBase:
+
+    CSV_DELIMITER = ";"
+    FILE_ENCODING = "utf-8"
+
+    def _get_header_row(self):
+        return [col.heading for col in self.CSV_COLUMNS]
+
+    def write_csv_file(self, path):
+        csv_string = self.get_csv_string()
+        with open(path, encoding=self.FILE_ENCODING, mode="w") as f:
+            f.write(csv_string)
+
+    def get_csv_string(self):
+        return self._make_csv(self.get_csv_lines())
+
+    def get_csv_lines(self):
+        lines = [self._get_header_row()]
+        for item in self.get_row_items():
```

Comment:
Could you please check, is this `get_row_items` method missing..?

Reply:
It's the responsibility of a derived class to provide that, like TalpaService and ApplicationsCsvService do

Reply:
Ah yes thank you. Maybe we can add these as abstract method / property..?